translib/path_utils.go

```diff
@@ -91,6 +97,16 @@ func NewPathInfo(path string) *PathInfo {
 
 		name := readUntil(r, '=')
 		value := readUntil(r, ']')
+
+		// Handle duplicate parameter names by suffixing "#N" to it.
+		// N is the number of occurance of that parameter name.
+		if info.HasVar(name) {
```

Comment:
It is not redundant and the "for" loop does not handle the intended case. This tool is for extracting parameter name-value map from the path. Yang does not mandate unique key attribute names in a yang tree. Hence parameter name can repeat in a path. Following is an example from the the new ietf-yang-library.yang.

/ietf-yang-library:modules-state/module[**name**=M1][**revision**=R1]/deviation[**name**=M2][**revision**=R2]

Since map object cannot have duplicate keys, we are adding a suffix for the repeated parameter names. The "if" statement checks whether a current parameter repeated. The for loop counts the number of repetitions so far.

Map contents for the above path will be {"name": "M1", "revision": "R1", "name#2": "M2", "revision#2": "R2"}

Reply:
I might have misinterpreted this comment. Yes, there will be one extra check if the param name is repeated. Since the HasVar is not a costly operation, I went for 1 extra check to keep the code compact (otherwise one more "if" check was required inside the loop -- there is no do-while loop in go). Anyway I have removed the top "if" condition now.